### PR TITLE
Visitor framework in commands hierarchy with tests

### PR DESCRIPTION
**Pull request by s218156**

### CODE REVIEW

**Comment by kowallus** [comment]
Where are classes with tests (e.g., SelectCommandVisitorCounterListener)? Maybe you have forgotten to commit them or push?

**Comment by s218156** [comment]
By mistake we made 2 commits with same title. Commit 4baf44c contains test classes.

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/CommandVisitorCounter.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import java.util.Iterator;
+
+public class CommandVisitorCounter implements CommandVisitorInterface {
+
+    private int operateToCommandCounter = 0;
+    private int setPositionCommandCounter = 0;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/CommandVisitorCounter.java]: Why counting different kinds of commands if we cannot access and use specific information.

File: src/test/java/edu/kis/powp/jobs2d/events/SelectCommandVisitorCounterListener.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        commands.add(new SetPositionCommand(0, -50));
+        commands.add(new OperateToCommand(0, -50));
+        commands.add(new SetPositionCommand(0, -40));
+        commands.add(new OperateToCommand(0, 50));
+        commands.add(new SetPositionCommand(70, -50));
+        commands.add(new OperateToCommand(20, -50));
+        commands.add(new OperateToCommand(20, 0));
+        commands.add(new OperateToCommand(70, 0));
+        commands.add(new OperateToCommand(70, 50));
+        commands.add(new OperateToCommand(20, 50));
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectCommandVisitorCounterListener.java]: Please move this command to a separate test so you can set it using driverCommandManager.

In this test use getCurrentCommand to access a command to be processed by the visitor.

**Review by kowallus — CHANGES_REQUESTED** [review]
Fine, but I need a minor update to make it more useful.

File: src/test/java/edu/kis/powp/jobs2d/events/SelectLoadTestCommandOptionListener.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+public class SelectLoadTestCommandOptionListener implements ActionListener {
+
+	@Override
+	public void actionPerformed(ActionEvent e) {
+		List<DriverCommand> commands = new ArrayList<DriverCommand>();
+		commands.add(new SetPositionCommand(-40, -50));
+		commands.add(new OperateToCommand(40, -50));
+		commands.add(new SetPositionCommand(0, -50));
+		commands.add(new OperateToCommand(0, 100));
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectLoadTestCommandOptionListener.java:22]: Wasn't the previous command longer?

**Review by kowallus — APPROVED** [review]
Thank you!

File: src/test/java/edu/kis/powp/jobs2d/events/SelectLoadTestCommandOptionListener.java (hunk 3 above)
Comment (s218156) [reply src/test/java/edu/kis/powp/jobs2d/events/SelectLoadTestCommandOptionListener.java:22]: It was copy of LoadSecretCommandOptionListener. To not double that we made another test figure (it draws "T" character)

**Pull request merged**